core/src/banking_stage/transaction_scheduler/scheduler_controller.rs

```diff
@@ -382,6 +402,29 @@ impl SchedulerController {
             );
         }
     }
+
+    /// Calculate prioritization for a transaction.
+    fn calculate_prioritization(
+        fee_structure: &FeeStructure,
+        include_loaded_account_data_size_in_fee: bool,
+        message: &SanitizedMessage,
+        budget_limits: &FeeBudgetLimits,
+        cost: u64,
+    ) -> u64 {
+        let total_fee = fee_structure.calculate_fee(
+            message,
+            5_000, // this just need to be non-zero
+            budget_limits,
+            include_loaded_account_data_size_in_fee,
+        );
+
+        // We need a multiplier here to avoid rounding down too aggressively.
+        // For many transactions, the cost will be greater than the fees in terms of raw lamports.
+        // For the purposes of calculating prioritization, we multiply the fees by a large number so that
+        // the cost is a small fraction.
+        const MULTIPLIER: u64 = 1_000_000;
+        total_fee.saturating_mul(MULTIPLIER).saturating_div(cost)
```

Comment:
nit: it seems `cost` can't be `0` as far as i read `.calculate_fee()` code. however, i think every `.*_div(cost)` or `/` needs an explicit commentary for the case of `0` or an explicit handling (maybe `0` priority in this case) if it's fed with user-controlled inputs.